src/persist/src/nemesis/direct.rs

```diff
@@ -52,14 +63,95 @@ struct Dataflow {
 // without graceful shutdown) and reimplement Direct using Indexed.
 #[derive(Debug)]
 pub struct Direct {
-    start_fn: StartFn,
-    pub persister: RuntimeClient,
+    workers: usize,
+    shared: Arc<DirectShared>,
+}
+
+#[derive(Debug)]
+struct DirectCore {
+    start_fn: Box<dyn StartRuntime>,
     unreliable: UnreliableHandle,
-    dataflows: HashMap<String, Dataflow>,
-    snapshots: HashMap<SnapshotId, DecodedSnapshot<String, ()>>,
+    persister: RuntimeClient,
+    dataflows: HashMap<String, (Ingest, Dataflow)>,
```

Comment:
its a little bit conceptually confusing to have both `Ingest` and `Dataflow` in one object because I thought that this pr was trying to split apart those two sides of the house? (I know this comment is ill-informed, vague, and unactionable sorry!)

Reply:
added an explanation of this to my big comment. this is an unfortunate side-effect of us not yet having read-only persist. once we have that, DirectCore will be split into an ingestd-side and a dataflowd-side. we could have them in separate maps in DirectCore in the meantime, which is actually how I originally wrote it, but there ended up being some code to handle a collection being in one but not the other and I couldn't figure out a way to make that less confusing than this is